src/main/java/com/mercateo/common/rest/schemagen/PropertyJsonSchemaMapper.java

```diff
@@ -75,12 +84,7 @@ private ObjectNode createPropertyEntry(JsonProperty jsonProperty, final ObjectNo
 
                 case STRING:
                     result.put("type", "string");
-                    if (hasAllowedValues(jsonProperty)) {
-                        result.set("enum", getAllowedValues(jsonProperty));
-                    }
-                    if (hasDefaultValue(jsonProperty)) {
-                        result.set("default", getDefaultValue(jsonProperty));
-                    }
+                    propertyJsonMapper.addStringDefaultAndAllowedValues(result, jsonProperty);
                     jsonProperty.getSizeConstraints().getMin().ifPresent(x -> result.put(
```

Comment:
Have you considered moving this logic into PropertyJsonMapper too? 
Have you considered using Polymorphism instead of switch statement?